src/ucs/debug/log.c

```diff
@@ -164,30 +173,40 @@ static void ucs_log_file_rotate()
 
     for (idx = ucs_log_file_last_idx - 1; idx >= 0; --idx) {
         ucs_log_get_file_name(old_log_file_name,
-                              sizeof(old_log_file_name), idx);
+                              PATH_MAX, idx);
         ucs_log_get_file_name(new_log_file_name,
-                              sizeof(new_log_file_name), idx + 1);
+                              PATH_MAX, idx + 1);
 
         if (access(old_log_file_name, W_OK) != 0) {
+            ucs_free(old_log_file_name);
+            ucs_free(new_log_file_name);
             ucs_fatal("unable to write to %s", old_log_file_name);
         }
```

Comment:
Printing log file names after freeing them seems like not a good idea. Also there is a lot of code duplication, so I would propose something like the following:

```suggestion
        if (access(old_log_file_name, W_OK) != 0) {
            ucs_error("unable to write to %s", old_log_file_name);
            goto err_out;
        }
        
...

err_out:
    ucs_free(old_log_file_name);
    ucs_free(new_log_file_name);
    if (err_msg) {
        ucs_fatal("ucs_log_file_rotate() fatal error (see logs before)");
    }
```

Reply:
Thank you :), please check now @ivankochin

Reply:
Outdated, done in other PRs